File: spring-data/jpa-pageable/src/main/java/techtabu/jpapageable/customer/Customer.java
package techtabu.jpapageable.customer;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

/**
 * @author TechTabu
 */

@Entity(name = "customer")
@Table(name = "customer")
@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
@Builder
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    private String customerNumber;
    private String email;
}

File: spring-data/jpa-pageable/src/main/kotlin/techtabu/jpapageable/customer/Customer.kt
package techtabu.jpapageable.customer

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table

/**
 * @author TechTabu
 */

@Entity(name = "customer")
@Table(name = "customer")
data class Customer (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int,

    var firstName : String,
    var lastName : String,
    var customerNumber: String,
    var email : String
)